fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1180,11 +1253,19 @@ def _free_fp16_param_shard(self, params: Optional[List[Parameter]] = None) -> No
                 p._fp16_shard.record_stream(current_stream)
                 free_storage_(p._fp16_shard)
 
-    def assert_state(self, state: TrainingState) -> None:
+    def assert_state(self, state: Union[TrainingState, List[TrainingState]]) -> None:
```

Comment:
made this assert an error checking since it seems to be useful even when people run python -O.